framework/contracts/native/account-factory/src/commands.rs

```diff
@@ -107,123 +105,124 @@ pub fn execute_create_account(
             ))
         })?;
     }
-    // save module for after-init check
-    CONTEXT.save(
-        deps.storage,
-        &Context {
-            account_id: account_id.clone(),
-            account_proxy_address: None,
-            manager_module: None,
-            proxy_module: Some(module.clone()),
-
-            additional_config: AdditionalContextConfig {
-                namespace,
-                base_asset,
-                name,
-                description,
-                link,
-                owner: governance.into(),
-            },
-            install_modules,
-            funds_for_install,
-        },
-    )?;
 
-    if let ModuleReference::AccountBase(proxy_code_id) = module.reference {
-        Ok(AccountFactoryResponse::new(
-            "create_account",
-            vec![
-                ("account_sequence", &account_id.seq().to_string()),
-                ("trace", &account_id.trace().to_string()),
-            ],
-        )
-        // Create proxy
-        .add_submessage(SubMsg {
-            id: CREATE_ACCOUNT_PROXY_MSG_ID,
-            gas_limit: None,
-            msg: WasmMsg::Instantiate {
-                code_id: proxy_code_id,
-                funds: funds_to_proxy.into_vec(),
-                // Currently set admin to self, update later when we know the contract's address.
-                admin: Some(env.contract.address.to_string()),
-                label: format!("Proxy of Account: {}", account_id),
-                msg: to_json_binary(&ProxyInstantiateMsg {
-                    account_id,
-                    ans_host_address: config.ans_host_contract.to_string(),
-                })?,
-            }
-            .into(),
-            reply_on: ReplyOn::Success,
-        }))
+    let salt = generate_salt(env.block.height, &account_id);
+
+    // Get code_ids
+    let (proxy_code_id, manager_code_id) = if let (
+        ModuleReference::AccountBase(proxy_code_id),
+        ModuleReference::AccountBase(manager_code_id),
+    ) = (
+        proxy_module.reference.clone(),
+        manager_module.reference.clone(),
+    ) {
+        (proxy_code_id, manager_code_id)
     } else {
-        Err(AccountFactoryError::WrongModuleKind(
-            module.info.to_string(),
+        return Err(AccountFactoryError::WrongModuleKind(
+            proxy_module.info.to_string(),
             "account_base".to_string(),
-        ))
-    }
-}
+        ));
+    };
 
-/// instantiates the Treasury contract of the newly created DAO
-pub fn after_proxy_create_manager(
-    deps: DepsMut,
-    env: Env,
-    result: SubMsgResult,
-) -> AccountFactoryResult {
-    let config = CONFIG.load(deps.storage)?;
+    // TODO: should be possible to parse checksums from artifacts/checksums.txt
```

Comment:
This code is only ran when an account is created so not that important to optimize.

Reply:
> This code is only ran when an account is created so not that important to optimize.

Pretty sure it will be a cool feature for `cosmwasm-std`!
Agree, it's not that important, but that would be a nice gas optimization, considering checksum will be queried for every account (should we save it at instantiation?)

Reply:
No, it's fine :)